src/io/flutter/sdk/FlutterSdk.java

```diff
@@ -61,6 +75,57 @@ public static FlutterSdk getFlutterSdk(@NotNull final Project project) {
     return project.isDisposed() ? null : getFlutterSdkByDartSdk(DartPlugin.getDartSdk(project));
   }
 
+  /**
+   * Returns the Flutter SDK for an unopened project that has a possibly broken "Dart SDK" project library.
+   * <p>
+   * (This can happen for a newly-cloned Flutter SDK where the Dart SDK is not cached yet.)
+   */
+  @Nullable
+  public static FlutterSdk getForProjectDir(final VirtualFile projectDir) {
+    final Library lib = loadDartSdkLibrary(projectDir);
+    if (lib == null) {
+      return null;
+    }
+
+
+    final String[] urls = lib.getUrls(OrderRootType.CLASSES);
+    for (String url : urls) {
+      if (url.endsWith(DART_CORE_SUFFIX)) {
+        final String flutterUrl = url.substring(0, url.length() - DART_CORE_SUFFIX.length());
+        final String flutterPath = getPathForProjectLibraryUrl(flutterUrl, projectDir);
+        return flutterPath == null ? null : new FlutterSdk(flutterPath);
+      }
+    }
+    return null;
+  }
+
+  @Nullable
+  private static Library loadDartSdkLibrary(@NotNull VirtualFile projectDir) {
+    final VirtualFile libPath = projectDir.findFileByRelativePath(SDK_LIBRARY_PATH);
+    if (libPath == null) {
+      return null;
+    }
+
+    try {
+      final Element elt = JDOMUtil.load(new File(libPath.getPath()));
+      final ProjectLibraryTable table = new ProjectLibraryTable();
+      table.readExternal(elt);
+      return table.getLibraryByName("Dart SDK");
```

Comment:
Getting library by name is ok (you may reuse `DartSdk.DART_SDK_LIB_NAME` constant). But the rest of the method is a bit hacky :). It's because you do not have `Project` instance in hands in `ProjectOpenProcessor`.

AFAIR normally `ProjectOpenProcessor` implementations extend `ProjectOpenProcessorBase` and do not override `doOpenProject()`. They work not more than once per project. `ProjectOpenProcessor` does its job when user opens a folder without `.idea` file inside - it helps to create initial project configuration. When you open existing project `ProjectOpenProcessor` normally do not do anything. If you need to check and fix anything in the opened project `StartupActivity` is there to help. 

I may not be remembering all the nuances, but I feel it is designed to work like this.